pkg/config/validation/validation.go

```diff
@@ -718,6 +718,16 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 					}
 				}
 			}
+
+			errs = appendErrors(errs, validateSidecarIngressTLS(i.InboundTls))
+
+			// If port is HTTPS or TLS, make sure that server has TLS options
+			p := protocol.Parse(i.Port.Protocol)
+			if p.IsTLS() && i.InboundTls == nil {
+				errs = appendErrors(errs, fmt.Errorf("sidecar: ingress must have TLS settings for HTTPS/TLS protocols"))
```

Comment:
hmm. I dont think this is going to work. I think people may have already defined Sidecars with appropriate ports and when they upgrade, this will cause breakage. So we might have to let this piece go. What we can check is if inbound TLS is defined, the port must be either TLS or HTTPS, and not HTTP or TCP.

Reply:
ok, understand